### PR TITLE
[CARBONDATA-1295] Optimize test cases,merge three QueryTest class into one

### PR DESCRIPTION
**Pull request by chenliang613**
1.Optimize test cases,merge three QueryTest class into one, and update related code.
2.Remove one invalid scalastyle.

### CODE REVIEW

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/440/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3028/

File: integration/spark2/src/test/scala/org/apache/spark/sql/common/util/Spark2QueryTest.scala [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+import org.apache.spark.sql.hive.CarbonSessionState
+import org.apache.spark.sql.test.util.QueryTest
+
+
+class Spark2QueryTest extends QueryTest {
+
+  val hiveClient = sqlContext.sparkSession.sessionState.asInstanceOf[CarbonSessionState]
+    .metadataHive
+
+  protected def createAndLoadInputTable(inputTableName: String, inputPath: String): Unit = {
```
Comment (ravipesala) [inline integration/spark2/src/test/scala/org/apache/spark/sql/common/util/Spark2QueryTest.scala]: Please remove this method from here, it should be in respective testcase

File: integration/spark2/src/test/scala/org/apache/spark/sql/common/util/Spark2QueryTest.scala [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+       """.stripMargin)
+
+    sql(
+      s"""
+         | LOAD DATA LOCAL INPATH '$inputPath'
+         | INTO TABLE $inputTableName
+       """.stripMargin)
+  }
+
+  protected def createAndLoadTestTable(tableName: String, inputTableName: String): Unit = {
```
Comment (ravipesala) [inline integration/spark2/src/test/scala/org/apache/spark/sql/common/util/Spark2QueryTest.scala]: Please remove this method from here, it should be in respective testcase

File: integration/spark2/src/test/scala/org/apache/spark/sql/common/util/Spark2QueryTest.scala (hunk 1 above)
Comment (chenliang613) [inline integration/spark2/src/test/scala/org/apache/spark/sql/common/util/Spark2QueryTest.scala]: ok

File: integration/spark2/src/test/scala/org/apache/spark/sql/common/util/Spark2QueryTest.scala (hunk 2 above)
Comment (chenliang613) [inline integration/spark2/src/test/scala/org/apache/spark/sql/common/util/Spark2QueryTest.scala]: ok

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/444/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3032/

**Comment by ravipesala** [comment]
LGTM

**Pull request closed without merging**